src/H5Edeprec.c

```diff
@@ -182,13 +182,29 @@ H5Eget_minor(H5E_minor_t min)
 herr_t
 H5Epush1(const char *file, const char *func, unsigned line, H5E_major_t maj, H5E_minor_t min, const char *str)
 {
-    herr_t ret_value = SUCCEED; /* Return value */
+    const char *tmp_file;            /* Copy of the file name */
+    const char *tmp_func;            /* Copy of the function name */
+    herr_t      ret_value = SUCCEED; /* Return value */
 
     /* Don't clear the error stack! :-) */
     FUNC_ENTER_API_NOCLEAR(FAIL)
 
+    /* Duplicate string information */
```

Comment:
Same as H5Epush2(): Duplicate strings and increment IDs for application error push operations.   Do this here, in the API routine, instead of within the library.